src/main/main.c

```diff
@@ -89,7 +165,7 @@ void opened() {
 void func_unlocking() {
     if (unlock_timer == 10 && !is_opened) {
         state = LOCKED;
-    } else if (unlock_timer < 10 && !is_opened) {
+    } else if (unlock_timer < 10 && !is_opened && is_authenticated()) {
```

Comment:
move this to bt connection

if (is_locked && !unlocking && !manual_lock && is_authenticated()) {
                unlocking = true;
                unlock_timer = 0;